cli/pkg/workspace/plugin/create.go

```diff
@@ -231,6 +235,52 @@ func (o *CreateWorkspaceOptions) Run(ctx context.Context) error {
 		return err
 	}
 
+	// Create context if --create-context is set
+	if o.CreateContextName != "" {
+		createContextOptions := NewCreateContextOptions(o.IOStreams)
+		createContextOptions.Name = o.CreateContextName
+		createContextOptions.ClientConfig = o.ClientConfig
+		createContextOptions.KeepCurrent = false
+
+		// If --enter is not set, do not switch to the new context
+		if !o.EnterAfterCreate {
+			createContextOptions.KeepCurrent = true
+		}
+
+		startingConfig, err := o.ClientConfig.RawConfig()
+		if err != nil {
+			return fmt.Errorf("failed to get raw config: %w", err)
+		}
+		createContextOptions.startingConfig = &startingConfig
+
+		// only for unit test needs
+		if o.modifyConfig != nil {
+			createContextOptions.modifyConfig = o.modifyConfig
+		}
+
+		if err := createContextOptions.Complete(nil); err != nil {
+			return err
+		}
+		if err := createContextOptions.Validate(); err != nil {
+			return err
+		}
+
+		err = createContextOptions.Run(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to create context: %w", err)
+		}
```

Comment:
```suggestion
		if err := createContextOptions.Run(ctx); err != nil {
			return fmt.Errorf("failed to create context: %w", err)
		}
```

Just a nit, lgtm otehrwise